MapboxNavigation/NavigationViewController.swift

```diff
@@ -531,7 +531,16 @@ extension NavigationViewController: RouteControllerDelegate {
     }
     
     public func routeController(_ routeController: RouteController, didArriveAt waypoint: Waypoint) {
-        guard routeController.routeProgress.isFinalLeg else { return }
+        guard routeController.routeProgress.isFinalLeg else {
```

Comment:
```swift
let isFinalLeg = routeController.routeProgress.isFinalLeg
let shouldIncrementLeg = delegate?.navigationViewController?(self, shouldIncrementLegWhenArrivingAtWaypoint: waypoint) ?? false

if !isFinalLeg || !shouldIncrementLeg {
    return delegate?.navigationViewController?(self, didArriveAt: waypoint)
}
```

IDK, this may be "6 one way, half-a-dozen the other"